polars/src/frame/mod.rs

```diff
@@ -1609,16 +1611,97 @@ impl Default for DataFrame {
     }
 }
 
+/// Conversion from Vec<RecordBatch> into DataFrame
+///
+/// If batch-size is small it might be advisable to call rechunk
+/// to ensure predictable performance
+impl std::convert::TryFrom<Vec<RecordBatch>> for DataFrame {
+    type Error = PolarsError;
+
+    fn try_from(batches: Vec<RecordBatch>) -> Result<DataFrame> {
+        // Validate that all record batches have the same schema
+        let mut batch_iter = batches.iter();
+
+        let first_batch = batch_iter.next().ok_or_else(|| {
+            PolarsError::NoData("At least one record batch is needed to create a dataframe".into())
+        })?;
+        let schema = first_batch.schema();
+        let fields = schema.fields();
+
+        for batch in batch_iter {
+            if batch.schema() != schema {
+                return Err(PolarsError::DataTypeMisMatch(
+                    "All record batches must have the same schema".into(),
+                ));
+            }
+        }
+
+        // Create a vector of series objects from record batches
+        let mut series_vec = Vec::with_capacity(fields.len());
+
+        for (field_idx, field) in Iterator::enumerate(schema.fields().iter()) {
+            let chunks: Vec<ArrayRef> = batches
+                .iter()
+                .map(|batch| batch.column(field_idx).clone())
+                .collect();
+            let series = Series::try_from((field.name().as_ref(), chunks))?;
+            series_vec.push(series);
+        }
+
+        DataFrame::new(series_vec)
```

Comment:
I believe recordbatches are guaranteed to have the same length and unique column names. If this is correct you can use `DataFrame::new_no_checks`. This skips the assertion of these requirements.

Reply:
`RecordBatch` guarantees same length columns but not unique column names. Do you think using `DataFrame::new` is approriate or would you manually check for unique column names instead?

The snippet below shows that you can create a `RecordBatch` where both columns are named `"a"`
```rust
extern crate arrow;

#[cfg(test)]
mod tests {

    use std::sync::Arc;
    use arrow::datatypes::{DataType, Schema, Field};
    use arrow::record_batch::RecordBatch;
    use arrow::array::{Int64Array, Float64Array};

    #[test]
    fn it_works() {

        let schema = Arc::new(Schema::new(vec![
            Field::new("a", DataType::Float64, false),
            Field::new("a", DataType::Int64, false)
        ]));

        let arr1 = Arc::new(Float64Array::from(vec![1.0, 2.0, 3.0, 4.0, 5.0]));
        let arr2 = Arc::new(Int64Array::from(vec![1, 2, 3, 4, 5]));

        RecordBatch::try_new(
            schema,
            vec![arr1, arr2]
        ).expect("Can create batch with same column name");

    }
}
```

Reply:
Then the DataFrame creation as is, is best indeed. :+1: